k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/PermanentFlagsResponse.java

```diff
@@ -22,10 +22,6 @@ private PermanentFlagsResponse(Set<Flag> flags, boolean canCreateKeywords) {
     }
 
     public static PermanentFlagsResponse parse(ImapResponse response) {
-        if (response.isTagged() || !equalsIgnoreCase(response.get(0), Responses.OK) || !response.isList(1)) {
```

Comment:
can this never happen? if it can't, maybe a `throw new IllegalArgumentException` would be appropriate?